### PR TITLE
fix: rpc get node info maybe crash when emqx_sys process is not start yet.

### PR DESCRIPTION
**Pull request by zhongwencool**
fix: rpc get node info maybe crash when other node not ready yet.
```erlang
2022-02-23T15:49:12.599555-05:00 [error] Minirest(Handler): Encode #{code => 0,
data => [{error,{'EXIT',{noproc,
{gen_server,call,[emqx_sys,uptime]}}}},#{datetime => <<"2022-02-23 15:49:12">>,
node => 'emqx@xx.com',node_status 
=> 'Running',otp_release => <<"24.1.5/12.1.5">>,
sysdescr => <<"EMQ X Enterprise">>,uptime => <<"57 minutes, 53 seconds">>,
version => <<"4.4.0">>},#{datetime => <<"2022-02-23 15:49:12">>,node => …

```

Refactor uptime/0  function: no longer dependent on `emqx_sys` process.

Possible problems with previous implementations: if `emqx_sys` process restarts, the uptime is Incorrect.

### CODE REVIEW

File: src/emqx_sys.erl [hunk 1]
```diff
@@ -94,7 +94,17 @@ sysdescr() -> emqx_app:get_description().
...
+    {TotalWallClock, _} = erlang:statistics(wall_clock),
+    {D, {H, M, S}} = calendar:seconds_to_daystime(TotalWallClock div 1000),
+    List = [{D, " days"}, {H, " hours"}, {M, " minutes"}, {S, " seconds"}],
+    {_, Uptime} =
+        lists:foldl(fun({Time, Unit}, {NeedSkip, Acc}) ->
+            case Time =:= 0 andalso NeedSkip of
+                true -> {NeedSkip, Acc};
+                false -> {false, [[integer_to_list(Time), Unit] |Acc]}
+            end
+                    end, {true, []}, List),
```
Comment (terry-xiaoyu) [inline src/emqx_sys.erl:106]: nit: bad indentions.

**Review by terry-xiaoyu — APPROVED** [review]

File: src/emqx_sys.erl [hunk 2]
```diff
@@ -94,7 +94,17 @@ sysdescr() -> emqx_app:get_description().
...
+    lists:flatten(lists:join(", ", lists:reverse(Uptime))).
```
Comment (zmstone) [inline src/emqx_sys.erl:107]: duplicated space? You have space in e,g. “ days” at line 100

Comment (zmstone) [reply src/emqx_sys.erl:107]: Nvm

File: src/emqx_sys.erl [hunk 3]
```diff
@@ -94,7 +94,17 @@ sysdescr() -> emqx_app:get_description().
 %% @doc Get sys uptime
 -spec(uptime() -> string()).
 uptime() ->
-    gen_server:call(?SYS, uptime).
+    {TotalWallClock, _} = erlang:statistics(wall_clock),
+    {D, {H, M, S}} = calendar:seconds_to_daystime(TotalWallClock div 1000),
+    List = [{D, " days"}, {H, " hours"}, {M, " minutes"}, {S, " seconds"}],
+    {_, Uptime} =
+        lists:foldl(fun({Time, Unit}, {NeedSkip, Acc}) ->
```
Comment (zmstone) [inline src/emqx_sys.erl:101]: lists:dropwhile/2 to delete the leading zeros, and use a lists:map to format.

File: src/emqx_sys.erl [hunk 4]
```diff
@@ -94,7 +94,17 @@ sysdescr() -> emqx_app:get_description().
 %% @doc Get sys uptime
 -spec(uptime() -> string()).
 uptime() ->
-    gen_server:call(?SYS, uptime).
+    {TotalWallClock, _} = erlang:statistics(wall_clock),
```
Comment (zmstone) [inline src/emqx_sys.erl:97]: Put this format function to a side-effect free func, so we can unit test the formatter

**Review by zmstone — DISMISSED** [review]

**Comment by zmstone** [comment]
Need to resolve conflict

**Review by zmstone — APPROVED** [review]

**Pull request merged**